contracts/staking/test/unit_tests/lib_proxy.ts

```diff
@@ -0,0 +1,324 @@
+                checkEndingConditions(
+                    await publicProxyCallAsync({
+                        calldata,
+                        revertRule: RevertRule.AlwaysRevert,
+                    }),
+                    false,
+                    calldata,
+                );
+            });
+
+            it('should revert with the correct data when the call falls and revertRule = AlwaysRevert', async () => {
+                const calldata = constructRandomFailureCalldata();
+
+                // Ensure that the returndata (the provided calldata) is correct.
+                checkEndingConditions(
+                    await publicProxyCallAsync({
+                        calldata,
+                        revertRule: RevertRule.AlwaysRevert,
+                    }),
+                    false,
+                    calldata,
+                );
+            });
+
+            it('should succeed with the correct data when the call succeeds and revertRule = NeverRevert', async () => {
+                const calldata = constructRandomSuccessCalldata();
+
+                // Ensure that the returndata (the provided calldata) is correct.
+                checkEndingConditions(
+                    await publicProxyCallAsync({
+                        calldata,
+                        revertRule: RevertRule.NeverRevert,
+                    }),
+                    true,
+                    calldata,
+                );
+            });
+
+            it('should succeed with the correct data when the call falls and revertRule = NeverRevert', async () => {
+                const calldata = constructRandomFailureCalldata();
+
+                // Ensure that the returndata (the provided calldata) is correct.
+                checkEndingConditions(
+                    await publicProxyCallAsync({
+                        calldata,
+                        revertRule: RevertRule.NeverRevert,
+                    }),
+                    true,
+                    calldata,
+                );
+            });
+
+            it('should succeed with the correct data when the call succeeds and revertRule = RevertOnError', async () => {
+                const calldata = constructRandomSuccessCalldata();
+
+                // Ensure that the returndata (the provided calldata) is correct.
+                checkEndingConditions(
+                    await publicProxyCallAsync({
+                        calldata,
+                    }),
+                    true,
+                    calldata,
+                );
+            });
+
+            it('should revert with the correct data when the call falls and revertRule = RevertOnError', async () => {
+                // Choose a random 4 byte string of calldata to send and replace the first byte with `0x00` to ensure
+                // that it does not call `publicProxyCall` by accident.
+                const calldata = '0x00'.concat(randomBytes4().slice(4, 10));
```

Comment:
Why aren't we using `constructRandomFailureCalldata()` here?

Reply:
Just an oversight